src/oscar/apps/partner/abstract_models.py

```diff
@@ -241,6 +246,8 @@ def consume_allocation(self, quantity):
     consume_allocation.alters_data = True
 
     def cancel_allocation(self, quantity):
+        if not self.product.product_class.track_stock:
```

Comment:
This logic is used in 4 places. I think we should move it into a separate property method on the stock record model called `should_manage_allocations()`. That makes it easier for a project to override the behaviour rather than changing it in 4 different places.

Reply:
@khink sorry, noticed one more issue regarding overridability. Would be great if you can update the PR to address this, and then we're good to merge.

Reply:
Not sure what you mean: I should think the `cancel/consume _stock_allocations` methods are easy enough to override if the need arises, so what should be made overridable? What scenario(s) do you have in mind?

Reply:
You are doing the same check [here](https://github.com/django-oscar/django-oscar/pull/2567/files#diff-0880376e680127dddc85b4b4f30f06edR196), in the order processing code. If you needed to override cancel/consume you would also have to override `order.processing.EventHandler` which I don't think is reasonable to expect someone to have to do. If we provide a wrapper method, then you can just override that and have `are_stock_allocations_available` work correctly.

Reply:
I've just seen your commit - it's all good now. Thanks!